sbws/core/scanner.py

```diff
@@ -503,13 +514,21 @@ def main_loop(args, conf, controller, relay_list, circuit_builder, result_dump,
                 [args, conf, destinations, circuit_builder, relay_list,
                  target], {}, callback, callback_err)
             pending_results.append(async_result)
-
+            measured_fp_set.add(async_result)
         # After the for has finished, the pool has queued all the relays
         # and pending_results has the list of all the AsyncResults.
         # It could also be obtained with pool._cache, which contains
         # a dictionary with AsyncResults as items.
         num_relays_to_measure = len(pending_results)
         wait_for_results(num_relays_to_measure, pending_results)
+        # NOTE: in a future refactor make State a singleton in __init__.py
```

Comment:
These three lines can go in the heartbeat module as well, right?

Reply:
Yeah, this three can go in the new module